### ISSUE TITLE
Multiple Lookup Filters example

### ISSUE THREAD

**Issue by ckelsoe** [volkanceylan/Serenity#1157]
Hi,

Is there an example with multiple quickfilters using lookup fields?

I am having trouble understanding how to have this code work for another  multiselect criteria. 

`
protected getQuickFilters() {
        let items = super.getQuickFilters();

        var genreListFilter = Q.first(items, x =>
            x.field == MovieRow.Fields.GenreList);

        genreListFilter.handler = h => {
            var request = (h.request as MovieListRequest);
            var values = (h.widget as Serenity.LookupEditor).values;
            request.Genres = values.map(x => parseInt(x, 10));
            h.handled = true;
        };

        return items;
`

**Comment by volkanceylan** [comment]
Is it so hard to replace field / entity names with yours.

**Comment by ckelsoe** [comment]
I really appreciate the helpful reply. 

When I add the second search item as show in the code ( WHERE I REPLACED THE FIELD NAMES) I get a blank grid with no search boxes at all. Obviously I am missing something. None of the examples have two multiselect field searches so I thought it reasonable to ask for an example that showed this functioning. 

This code:

`

namespace PSCRM {
    
    @Serenity.Decorators.registerClass()
    export class CustomerGrid extends Serenity.EntityGrid<CustomerRow, any> {
        protected getColumnsKey() { return 'Customer'; }
        protected getDialogType() { return CustomerDialog; }
        protected getIdProperty() { return CustomerRow.idProperty; }
        protected getLocalTextPrefix() { return CustomerRow.localTextPrefix; }
        protected getService() { return CustomerService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }

        protected getQuickFilters() {
            let items = super.getQuickFilters();

            var AssociationFilter = Q.first(items, x =>
                x.field == CustomerRow.Fields.AssociationList);

            AssociationFilter.handler = h => {
                var request = (h.request as CustomerListRequest);
                var values = (h.widget as Serenity.LookupEditor).values;
                request.Associations = values.map(x => parseInt(x, 10));
                h.handled = true;

                // Trying to add additional quick filters here
                var ProductFilter = Q.first(items, x =>
                    x.field == CustomerRow.Fields.ProductList);
                
                ProductFilter.handler = p => {
                    var request = (p.request as CustomerListRequest);
                    var values = (p.widget as Serenity.LookupEditor).values;
                    request.Associations = values.map(x => parseInt(x, 10));
                    h.handled = true;

            };

            return items;
        }

        protected getQuickSearchFields():
            Serenity.QuickSearchField[] {
            let fld = CustomerRow.Fields 
            return [
                { name: "", title: "all" },
                { name: fld.CustomerName, title: "customer" },
                { name: fld.XxxxxforeId, title: "xxxxfore id" }
            ];
        }
    }
}
`

Results in this: 
![image](https://cloud.githubusercontent.com/assets/21282091/18367640/33189b94-75ea-11e6-9002-a1359f942039.png)

instead of this with an additional search box (of which does not work because it has not been connected to the field which the code above attempts to do):
![image](https://cloud.githubusercontent.com/assets/21282091/18367892/3602f56a-75eb-11e6-9adb-7535857b8a74.png)

The Industry Associations search does work because I was able to change the field names as you have suggested in your earlier reply. 
 
I have the correct field name in the second search, "CustomerRow.Fields.ProductList". What would cause the display results. That is what I have been unable to figure out.

**Comment by volkanceylan** [comment]
If you expect more helpful replies you should report more helpful issues. From your first question, its impossible to understand what your issue is unless you have some magic sphere. 

You are trying to create a second quick filter inside handler of first one, why? 

Second, you should look at browser console first, to see if an error is there.

**Comment by ckelsoe** [comment]
The first post was just simply asking for an example of how to create a second filter when there is more than one multiselect field that filtering is required for. Obviously asking for an example must have been the wrong thing to do or perhaps I was not clear enough in that request. 

I do not understand where the second filter setup should occur. I put it where I did in the code because it seemed that getQuickFilter**s** was named in the plural because it would be where all of that type of filtering mechanism would be created.  I am trying with replicating the getQuickFilters section by making a copy and renaming it to getQuickFilters1. That does eliminate the issue reported above. I do have another issue that I am working through currently with a field name not being found. I used the same field name CustomerRow.Fields.ProductList listed above which also resides in CustomerRow.cs, however, the process errors out because the field name cannot be found. Once I figure out why it cannot find the field that is defined in CustomerRow.cs then I should be good to go.

I do very much appreciate the huge effort you have put into this product and your willingness to help the community. I am not a C# developer so I have a lot of learning to do. Your product is helping with that. Part of the frustration for you must be questions that an experienced C# developer would know whereas I have to ask. I am sorry for that on my part.

**Comment by ckelsoe** [comment]
At this point I am assuming that there cannot be two multivalue lookup fields in the quick search. The error is that the field name is incorrect, however, it is the same field name everywhere it is referenced. I do not know if I need to have new code in the endpoint.cs and repository.cs - if so what I have tried does not work. 

In my code the field "ProductList" has been created in CustomerRow.cs, Is also in CustomerColumns.cs, in CustomerForm.cs, in CustomerGrid.ts.

I added it below the other multiselect field in CustomerListRequest.cs, however that did not make a difference

I do not know where else to look.
